sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/policy/ProviderRegistrationPolicy.java

```diff
@@ -67,16 +69,9 @@ private boolean isResponseSuccessful(HttpResponse response) {
                             }
 
                             if (cloudError != null && MISSING_SUBSCRIPTION_REGISTRATION.equals(cloudError.getCode())) {
-                                String subscriptionId = ResourceUtils.extractFromResourceId(
-                                    request.getUrl().getPath(), "subscriptions");
-                                RestClient restClient = new RestClientBuilder()
-                                        .withBaseUrl(String.format("%s://%s",
-                                            request.getUrl().getProtocol(), request.getUrl().getHost()))
-                                        .withCredential(credential)
-                                        .withSerializerAdapter(jacksonAdapter).buildClient();
                                 // TODO: add proxy in rest client
-                                ResourceManager resourceManager = ResourceManager.authenticate(restClient)
-                                        .withSubscription(subscriptionId);
+                                ResourceManager resourceManager = ResourceManager.authenticate(credential, profile)
+                                        .withDefaultSubscription();
```

Comment:
same as above, if profile, why defaultsubs? what's defaultsubs?

Reply:
At first step, the users can create instance for `ResourceManager.Authenticated` without subscription id. Next step, they can use `withSubscription(id)` pass subs id into profile. In another case, they can use `withDefaultSubscription()` if profile already contains subs id.